src/main/webapp/app/admin/system-notification-management/system-notification-management-update.component.ts

```diff
@@ -35,9 +35,15 @@ export class SystemNotificationManagementUpdateComponent implements OnInit {
 
     /**
      * Returns to previous state (same as back button in the browser)
+     * Returns to the detail page if there is no previous state and we edited an existing notification
+     * Returns to the overview page if there is no previous state and we created a new notification
      */
     previousState() {
-        window.history.back();
+        if (window.history.length > 1) {
```

Comment:
it seems this pattern is repeated several times. Can we extract a helper function? We could pass the new url (e.g. `['../']`) to navigate in it to make it usable for different places